### PR TITLE
add support for hongkong

### PR DESCRIPTION
**Pull request by M00nF1sh**
### Description
Add support for new region

### Checklist

- [x] Code compiles correctly (i.e `make build`)
- [x] Added tests that cover your change (if possible)
- [x] All unit tests passing (i.e. `make test`)
- [x] Added/modified documentation as required (such as the `README.md`, and `examples` directory)
- [x] Manually tested

### CODE REVIEW

File: pkg/nodebootstrap/assets/10-eksclt.al2.conf [hunk 1]
```diff
@@ -20,6 +20,6 @@ ExecStart=/usr/bin/kubelet \
   --network-plugin=cni \
   --cni-bin-dir=/opt/cni/bin \
   --cni-conf-dir=/etc/cni/net.d \
-  --pod-infra-container-image=602401143452.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/eks/pause-amd64:3.1 \
+  --pod-infra-container-image=${AWS_EKS_ECR_ACCOUNT}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/eks/pause-amd64:3.1 \
```
Comment (errordeveloper) [inline pkg/nodebootstrap/assets/10-eksclt.al2.conf]: please also see `pkg/nodebootstrap/assets/bootstrap.ubuntu.sh`

File: pkg/nodebootstrap/userdata.go [hunk 2]
```diff
@@ -24,6 +24,9 @@ import (
 const (
 	configDir            = "/etc/eksctl/"
 	kubeletDropInUnitDir = "/etc/systemd/system/kubelet.service.d/"
+
+	eksECRAccountStandard = "602401143452"
```
Comment (errordeveloper) [inline pkg/nodebootstrap/userdata.go]: We use this account ID for searching AMIs also, should we parametrised it everywhere?

https://github.com/weaveworks/eksctl/blob/e9fc98eef14a8bbe86f417ba10b56b9bb438964c/pkg/ami/auto_resolver.go#L53

**Comment by errordeveloper** [comment]
Also, AMIs would need to be added also. If `602401143452` owns the AMIs, `make generate-ami` should be sufficient.

File: pkg/nodebootstrap/userdata.go (hunk 2 above)
Comment (M00nF1sh) [reply pkg/nodebootstrap/userdata.go]: Thanks for the info :D. I'll update here as well. For opt-in regions, we have to use another account to host these resources. ami/ecr..

File: pkg/nodebootstrap/assets/10-eksclt.al2.conf (hunk 1 above)
Comment (M00nF1sh) [reply pkg/nodebootstrap/assets/10-eksclt.al2.conf]: updated :D

**Comment by M00nF1sh** [comment]
@errordeveloper Hi, do you know how can i trigger the addons updates? would like to manual test these code branches as well :D

**Comment by M00nF1sh** [comment]
The amiIDs will only be public at the time of actual launch.

**Comment by errordeveloper** [comment]
@M00nF1sh did you see https://eksctl.io/usage/cluster-upgrades/?

**Comment by martina-if** [comment]
Hi @M00nF1sh I think maybe if you create a cluster with version 1.12 then you might be able to test add-ons upgrades?

By the way, do you have time to rebase and finish this PR this week?

File: pkg/ami/auto_resolver.go [hunk 3]
```diff
@@ -48,10 +49,13 @@ var ImageSearchPatterns = map[string]map[string]map[int]string{
...
 
-// ImageFamilyToAccountID is a map of image families to account Ids
-var ImageFamilyToAccountID = map[string]string{
-	ImageFamilyAmazonLinux2: "602401143452",
-	ImageFamilyUbuntu1804:   "099720109477",
+// accountIDResolverFunc resolves ami provider accountID by region
+type accountIDResolverFunc func(region string) string
+
+// ImageFamilyToAccountID is a map of image families to accountIDResolverFunc
+var ImageFamilyToAccountID = map[string]accountIDResolverFunc{
```
Comment (martina-if) [inline pkg/ami/auto_resolver.go]: Perhaps now that we need a function anyway it makes more sense to get rid of the map and make it a function:

```
accountId(imageFamily, region) (string, error) {
  switch imageFamily {
    case ImageFamilyUbuntu1804:
      return "099720109477", nil
    case ImageFamilyAmazonLinux2:
      return api.EKSResourceAccountID(region), nil
    default:
      return nil, new error....
  }
}
```

Comment (martina-if) [reply pkg/ami/auto_resolver.go]: Then we can unit test it

Comment (M00nF1sh) [reply pkg/ami/auto_resolver.go]: make sense :D, modified

**Comment by M00nF1sh** [comment]
@martina-if 
I have rebased and test the upgrade addons, it works :D.
Actually the upgrade addons have other problems(but not in the scope of this PR):
1. the kube-proxy minor version may not be same as CP version. (we don't always upgrade it for new k8s minor versions).
1. the awsNode version is not tied to specific k8s major version. e.g. the newest awsNode in 1.12 have 1.5.0 as CNI version, eksctl upgrade will actually downgrade it to 1.4.1 currently.

**Comment by martina-if** [comment]
Hi @M00nF1sh , thanks for the fixes and for pointing those issues out . We already have a PR for nr 2 :)

**Comment by errordeveloper** [comment]
> the upgrade addons have some problems

Please open an issue for each of these items, and we will discuss there.

**Comment by gemagomez** [comment]
2) has a PR already that we are about to land: https://github.com/weaveworks/eksctl/pull/945
1) needs a new issue.

**Comment by martina-if** [comment]
closes https://github.com/weaveworks/aws-workstream/issues/43

**Pull request merged**